infra/src/test/kotlin/fi/fta/geoviite/infra/TestDBService.kt

```diff
@@ -0,0 +1,379 @@
+                "plan_project",
+                "plan_project_version",
+                "plan_author",
+                "plan_author_version",
+                "plan_version",
+                "switch",
+                "switch_joint",
+                "vertical_intersection",
+            )
+        )
+    }
+
+    fun clearPublicationTables() {
+        deleteFromTables(
+            schema = "publication",
+            tables = arrayOf(
+                "km_post",
+                "location_track",
+                "location_track_km",
+                "publication",
+                "reference_line",
+                "switch",
+                "switch_joint",
+                "switch_location_tracks",
+                "track_number",
+                "track_number_km",
+            ),
+        )
+    }
+
+    fun clearSplitTables() {
+        deleteFromTables(
+            schema = "publication",
+            tables = arrayOf(
+                "split",
+                "split_version",
+                "split_relinked_switch",
+                "split_relinked_switch_version",
+                "split_target_location_track",
+                "split_Target_location_track_version",
+            ),
+        )
+    }
+
+    fun clearProjektivelhoTables() {
+        deleteFromTables(
+            schema = "projektivelho",
+            tables = arrayOf(
+                "document_content",
+                "document_rejection",
+                "document",
+                "document_type",
+                "material_category",
+                "material_group",
+                "material_state",
+                "technics_field",
+                "project_state",
+                "assignment",
+                "project",
+                "project_group",
+                "search",
+            ),
+        )
+    }
+
+    fun clearRatkoTables() {
+        deleteFromTables(
+            schema = "integrations",
+            tables = arrayOf(
+                "ratko_push_content",
+            ),
+        )
+    }
+
+    fun getUnusedTrackNumber(): TrackNumber {
+        return TrackNumber(getUniqueName(DbTable.LAYOUT_TRACK_NUMBER, trackNumberLength.last))
+    }
+
+    fun getUnusedSwitchName(): SwitchName {
+        return SwitchName(getUniqueName(DbTable.LAYOUT_SWITCH, switchNameLength.last))
+    }
+
+    fun getUnusedProjectName(): ProjectName = ProjectName(
+        getUniqueName(DbTable.GEOMETRY_PLAN_PROJECT, 100)
+    )
+
+    fun getUnusedAuthorCompanyName(): CompanyName = CompanyName(
+        getUniqueName(DbTable.GEOMETRY_PLAN_AUTHOR, 100)
+    )
+
+    fun getUnusedBulkTransferId(): IntId<BulkTransfer> {
+        return getUniqueId(DbTable.PUBLICATION_SPLIT, "bulk_transfer_id")
+    }
+
+    private fun getUniqueName(table: DbTable, maxLength: Int): String {
+        val sql = "select max(id) max_id from ${table.versionTable}"
+        val maxId = jdbc.queryForObject(sql, mapOf<String, Any>()) { rs, _ -> rs.getInt("max_id") }!!
```

Comment:
Nämä ei taida toimia tyhjällä kannalla, vai? 🤔 Niitä tilanteita varten voisi olla järkevää kehitellä vaikka joku default-arvo, joka silloin aina annetaan

Reply:
Hmm... jos kanta on tyhjä niin `max` palauttanee null, eli kyllä sieltä tulee paluurivi kuitenkin -> `queryForObject` pitäisi toimia. `ResultSet.getInt` on JDBC:ssä niinkin kiero että palauttaa 0 jos arvo oli null, joten kyllä tuon pitäisi toimia. Luulisin että meillä aika moni testi failaisi ekalla ajolla jos näin ei olisi, sillä tätähän käytetään varsin paljon.